awx/main/models/inventory.py

```diff
@@ -209,6 +209,70 @@ def get_group_children_map(self):
             group_children.add(from_group_id)
         return group_children_map
 
+    def get_script_data(self, data, hostvars=False, show_all=False):
+        if show_all:
+            hosts_q = dict()
+        else:
+            hosts_q = dict(enabled=True)
+
+        if self.variables_dict:
+            all_group = data.setdefault('all', dict())
+            all_group['vars'] = self.variables_dict
+        if self.kind == 'smart':
+            if len(self.hosts.all()) == 0:
+                return {}
+            else:
+                all_group = data.setdefault('all', dict())
+                smart_hosts_qs = self.hosts.all()
+                smart_hosts = list(smart_hosts_qs.values_list('name', flat=True))
+                all_group['hosts'] = smart_hosts
```

Comment:
For both here, and the similar line a little lower, the more modern approach will be to put them in "ungrouped", and with our current Ansible support matrix, I think we can do this just about any time.

Reply:
👍 let's tackle that in another PR